lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -382,6 +379,60 @@ def test_get_get_errors(self, server_config, query_as):
             "total": 3,
         }
 
+    def test_use_funk_metalog_keys(self, query_as):
+        """Test funky metadata.log keys
+
+        Normally we constrain metadata keys to lowercase alphanumeric strings.
+        Traditional Pbench Agent `metadata.log` files contain keys constructed
+        from benchmark iteration values that can contain mixed case and symbol
+        characters. We allow these keys to be filtered and retrieved, but not
+        created, so test that we can filter on a funky key value and return
+        the key.
+
+        Args:
+            query_as: Query helper fixture
+        """
+        fio_1 = Dataset.query(name="fio_1")
+        Metadata.create(
+            dataset=fio_1,
+            key=Metadata.METALOG,
+            value={
+                "pbench": {
+                    "date": "2020-02-15T00:00:00",
+                    "config": "test1",
+                    "script": "unit-test",
+                    "name": "fio_1",
+                },
+                "iterations/fooBar=10-what_else@weird": {
+                    "iteration_name": "fooBar=10-what_else@weird"
+                },
+                "run": {"controller": "node1.example.com"},
+            },
+        )
+        response = query_as(
+            {
+                "metadata": "dataset.metalog.iterations/fooBar=10-what_else@weird",
+                "filter": "dataset.metalog.iterations/fooBar=10-what_else@weird.iteration_name:~10",
```

Comment:
Do we guard against colons in metadata keys?....

Reply:
If we take as given that colons are not allowed in keys, then the first colon **must** be at the end of the key, and any subsequent colon must be embedded in the value.  Thus, the "simple `.split(, maxsplit=1)`" is the appropriate implementation.  Using `rsplit()` would (in the case of an embedded colon) yield an impossible key (with an embedded colon) and a fractured value (I think...).

But, my question here is, if we are permitting "weird" values in Agent-generated (and Agent-supported user-supplied?) values, then embedded colons seem like a likely thing to have (e.g., `EC2::`)...but, perhaps only in values and not in keys?...  Hence my question about "guarding"....